atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraLogHelper.java

```diff
@@ -81,23 +89,38 @@ public static List<String> tokenRangeHashes(Set<TokenRange> tokenRanges) {
                 .collect(Collectors.toList());
     }
 
-    // Weakly intern instances as there should be a relatively small, generally fixed number of Cassandra servers.
-    @Value.Immutable(lazyhash = true, intern = true, builder = false)
-    @WeakInterningImmutablesStyle
-    interface HostAndIpAddress {
+    @Value.Immutable(lazyhash = true, builder = false, copy = false)
+    @JsonDeserialize(as = ImmutableHostAndIpAddress.class)
+    @JsonSerialize(as = ImmutableHostAndIpAddress.class)
+    abstract static class HostAndIpAddress {
         @Value.Parameter
-        String host();
+        abstract String host();
 
         @Nullable
         @Value.Parameter
-        String ipAddress();
+        abstract String ipAddress();
 
-        static HostAndIpAddress fromAddress(InetSocketAddress address) {
-            InetAddress inetAddress = address.getAddress();
-            if (inetAddress != null) {
-                return ImmutableHostAndIpAddress.of(address.getHostString(), inetAddress.getHostAddress());
+        @JsonIgnore
+        @Value.Lazy
+        String asString() {
+            if (ipAddress() == null) {
+                // unresolved IP
+                return host();
             }
-            return ImmutableHostAndIpAddress.of(address.getHostString(), /* unresolved IP */ null);
+            return host() + '/' + ipAddress();
```

Comment:
we're now interning `HostAndIpAddress` instances and lazily computing the string form to avoid extra allocations when logging & usage in thread name